libcudacxx/include/cuda/std/detail/libcxx/include/__config

```diff
@@ -1439,11 +1439,20 @@ typedef __char32_t char32_t;
 #endif
 #endif
 
+// We want to hide defaulted SMF but those cannot be marked as __host__ __device__
```

Comment:
This is the main change. We use the `_CCCL_ATTRIBUTE_HIDDEN` attribute which also suppresses warnings and either go to internal linkage or `_LIBCUDACXX_EXCLUDE_FROM_EXPLICIT_INSTANTIATION`

One potentially controversial decision is to pull `inline` into the macro. The reason being that we need it in case we are falling back to `_LIBCUDACXX_ALWAYS_INLINE`, e.g. for gcc which does not supports the attributes clang uses

Reply:
Does this break anything while compiling libcxx tests?

Reply:
I did not test it but the issue arose from libc++ marking defaulted constructors with `_LIBCPP_HIDE_FROM_ABI`